tests/test_wallet.py

```diff
@@ -985,6 +985,14 @@ def test_hsm_secret_encryption(node_factory):
     assert id == l1.rpc.getinfo()["id"]
 
 
+class HsmTool(TailableProc):
+    """Helper for testing the hsmtool as a subprocess"""
+    def __init__(self, *args):
+        TailableProc.__init__(self)
+        assert hasattr(self, "env")
+        self.cmd_line = ["tools/hsmtool", *args]
```

Comment:
This will clash with #4303, so merging either will require a rebase :-)

Reply:
Yeah, i worked on #4303 first but didn't want it to block the fix for whatever reason :)

Reply:
No problem, the changes just seemed eerily familiar xD